rust-semantics/expression/loops.md

```diff
@@ -0,0 +1,31 @@
+```k
+
+module RUST-LOOP-EXPRESSIONS
+    imports RUST-SHARED-SYNTAX
+    imports RUST-VALUE-SYNTAX
+    imports INT
+    imports MINT
+
+    syntax IteratorLoopExpression ::=  "for1" Pattern "in" ExpressionExceptStructExpression BlockExpression
+                                    | "for2" Pattern "in" ExpressionExceptStructExpression BlockExpression
+
+    rule for Patt:Identifier:PatternNoTopAlt | R:PatternNoTopAlts in First..Last B:BlockExpression => 
+            {
+                .InnerAttributes
+                (for1 Patt:Identifier:PatternNoTopAlt | R:PatternNoTopAlts in First..Last B:BlockExpression):IteratorLoopExpression; // Covers the cases of "for x | x in range" 
+                .NonEmptyStatements
+            };
+
+    rule for1 Patt:Identifier:PatternNoTopAlt | .PatternNoTopAlts in First..Last B:BlockExpression => 
+                let Patt = First; ~>
+                if (Patt :: .PathExprSegments):PathExprSegments < Last { .InnerAttributes B; (for2 Patt:Identifier:PatternNoTopAlt | .PatternNoTopAlts in First..Last B):IteratorLoopExpression;  .NonEmptyStatements};
+
+    rule for2 Patt:Identifier:PatternNoTopAlt | .PatternNoTopAlts in _..Last B:BlockExpression => 
+            let Patt = (Patt :: .PathExprSegments):PathExprSegments + ptrValue(null, u64(Int2MInt(1:Int)));
```

Comment:
Hmmm... This is interesting...

Right now, we have defined addition only between ints of the same kind, and I think that Rust does not allow adding ints of different types, e.g. `12u64 + 1u32` does not compile. Now, this probably works for our examples, but it's not correct, so here are some suggestions:
* Add a TODO saying that this is wrong and it should be fixed.
* define and use a function `intOfSameType(1, Last)` (Last should be a value when this rule is being applied, so it should have a concrete int type)